### ISSUE TITLE
DMSF main page not opening for a few users (Error 500)

### ISSUE THREAD

**Issue by naveedkamran** [danmunn/redmine_dmsf#1075]
DMSF main page is not opening for some users because of some error. The logs of error are here:

```
[2020-01-16T16:07:00.213540 #1]  INFO -- : Sent file /usr/src/redmine/tmp/thumbnails/4332_ee070e5c9782102adfa56e51f0d2dbf01e9f32d13f4723e4a0cac26bd489e3a4_64.thumb (0.1ms)
I, [2020-01-16T16:07:00.213672 #1]  INFO -- : Completed 200 OK in 89ms (ActiveRecord: 74.2ms)
I, [2020-01-16T16:07:03.750362 #1]  INFO -- : Started GET "/projects/tsmain/dmsf" for 172.17.0.1 at 2020-01-16 16:07:03 +0100
I, [2020-01-16T16:07:03.751144 #1]  INFO -- : Processing by DmsfController#show as HTML
I, [2020-01-16T16:07:03.751183 #1]  INFO -- :   Parameters: {"id"=>"tsmain"}
I, [2020-01-16T16:07:03.807869 #1]  INFO -- :   Current user: sebastian (id=75)
I, [2020-01-16T16:07:03.828680 #1]  INFO -- :   Rendering plugins/redmine_dmsf/app/views/dmsf/show.html.erb within layouts/base
I, [2020-01-16T16:07:03.830335 #1]  INFO -- :   Rendered plugins/redmine_dmsf/app/views/dmsf/_path.html.erb (0.2ms)
I, [2020-01-16T16:07:03.856496 #1]  INFO -- :   Rendered plugins/redmine_dmsf/app/views/dmsf/_dir.html.erb (13.3ms)
I, [2020-01-16T16:07:03.858245 #1]  INFO -- :   Rendered plugins/redmine_dmsf/app/views/dmsf/_dmsf_rows.erb (24.8ms)
I, [2020-01-16T16:07:03.858303 #1]  INFO -- :   Rendered plugins/redmine_dmsf/app/views/dmsf/_tree_view.erb (26.3ms)
W, [2020-01-16T16:07:03.861830 #1]  WARN -- : public/plugin_assets/redmine_dmsf/javascripts/jquery.dataTables/locales/en-GB.json not found
I, [2020-01-16T16:07:03.864217 #1]  INFO -- :   Rendered plugins/redmine_dmsf/app/views/dmsf/show.html.erb within layouts/base (35.5ms)
I, [2020-01-16T16:07:03.864384 #1]  INFO -- : Completed 500 Internal Server Error in 113ms (ActiveRecord: 70.2ms)
F, [2020-01-16T16:07:03.865308 #1] FATAL -- :   
F, [2020-01-16T16:07:03.865340 #1] FATAL -- : ActionView::Template::Error (no implicit conversion of true into String):
F, [2020-01-16T16:07:03.865515 #1] FATAL -- :     153:               details: false
    154:           },
    155:           language: {
    156:               url: "<%= plugin_asset_path(:redmine_dmsf, 'javascripts', json_url) %>"
    157:           },
    158:           oLanguage: {
    159:             sSearch: "<%= l(:description_filter) %>:"
F, [2020-01-16T16:07:03.865539 #1] FATAL -- :   
F, [2020-01-16T16:07:03.865557 #1] FATAL -- : plugins/redmine_dmsf/app/helpers/dmsf_helper.rb:76:in `join'
plugins/redmine_dmsf/app/helpers/dmsf_helper.rb:76:in `plugin_asset_path'
plugins/redmine_dmsf/app/views/dmsf/show.html.erb:156:in `block (2 levels) in _plugins_redmine_dmsf_app_views_dmsf_show_html_erb__3904826373779818733_47184814259480'
plugins/redmine_dmsf/app/views/dmsf/show.html.erb:149:in `block in _plugins_redmine_dmsf_app_views_dmsf_show_html_erb__3904826373779818733_47184814259480'
plugins/redmine_dmsf/app/views/dmsf/show.html.erb:133:in `_plugins_redmine_dmsf_app_views_dmsf_show_html_erb__3904826373779818733_47184814259480'
plugins/redmine_dmsf/app/controllers/dmsf_controller.rb:64:in `block (2 levels) in show'
plugins/redmine_dmsf/app/controllers/dmsf_controller.rb:62:in `show'
lib/redmine/sudo_mode.rb:63:in `sudo_mode'
plugins/redmine_dmsf/lib/redmine_dmsf/webdav/custom_middleware.rb:68:in `call'
```

**Comment by picman** [comment]
1. The users have got set the language to en-GB?
2. What is DMSF and Redmine version?

**Comment by naveedkamran** [comment]
Yes, I checked both the users have set their Language to English (British). 
Redmine version 4.0.4.stable
DMSF version 2.4.1

**Comment by naveedkamran** [comment]
So, I confirmed that users who have selected "en-GB" as their language are having issue. Setting their language to EN resolves the problem. US English date format is confusing for some users and they set their language to English GB.

**Comment by picman** [comment]
Hm, it's strange. I can't find any reason for this error. It seems that the helper method _json_url_ returns false instead of a string and then _plugin_asset_path_ method fails. But, how can _json_url_ return false it's a mystery for me.

**Comment by picman** [comment]
Fixed in #1080.
